### PR TITLE
Deploy this server automatically

### LINKED ISSUE #6: Set up this repository to auto-deploy to the server
This is a follow-up on https://github.com/whatwg/misc-server/pull/53.

After autodeploy we should use pm2 to restart.

Ideally we should autodeploy private-config.json from a private repository as well? That can be done later though.

@foolip, I was thinking of just setting up a Travis file that does rsync on the file contents and then uses SSH to execute the restart command. The SSH key would be encrypted in the same way we do for specs. Any thoughts or concerns?

### PR DESCRIPTION
**Pull request by domenic**
Fixes #6.

This is missing the actual encoded deploy key; @foolip can hopefully help me figure out how to generate that. (I'm not sure how the noderunner user is set up.)

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+chmod 600 deploy_key
+eval "$(ssh-agent -s)"
+ssh-add deploy_key
+
+# Do the deploy, first with rsync, then by SSHing in to restart
+
+echo "$SERVER $SERVER_PUBLIC_KEY" > known_hosts
+
+rsync --rsh="ssh -o UserKnownHostsFile=known_hosts" --verbose
+      --include-from=.deployinclude --delete \
```
Comment (foolip) [inline deploy.sh]: The `--delete` will cause the private-config.json to be deleted.

Comment (domenic) [reply deploy.sh]: Even in combination with --include? Sigh, the --include/--exclude/--exclude-delete options are quite a mess. I was hoping to do a test run, but it's a bit tricky without the private key.

Comment (foolip) [reply deploy.sh]: Hmm, I'm actually not sure without testing.

**Comment by domenic** [comment]
OK, well that was a fun adventure into the joys of rsync configuration, npm bugs, bash escaping, and more. Seems to work now!

Optional things to do before merging:

- Also deploy private-config.json. Probably easiest to just encode it?
- Should we be chmodding these files as we rsync them?

**Pull request merged**